lib/elements/dom-repeat.html

```diff
@@ -554,6 +564,7 @@
       for (let i=this.__instances.length-1; i>=instIdx; i--) {
         this.__detachAndRemoveInstance(i);
       }
+      this._setFilteredItemCount(isntIdxToItemsIdx.length);
```

Comment:
Would it make sense to gate setting the filtered item count on actually having a filter applied (and documenting it that way)?   If you don't have a filter, the total eventual count is just `items.length`, and this would avoid the perf impact of firing a notifying event every render for the (majority) case where no filter is applied.  Or is that too confusing?

Reply:
Is the perf impact likely to be noticeable?

By only doing setting `filteredItemCount` when a `filter` is applied, we risk pushing some logic into user-space. For example, if I have a repeat that conditionally has a `filter` function, then I'll need my component to conditionally choose whether to use `filteredItemCount` or `items.length` based on whether or not `filter` is set.